mlir/lib/Dialect/Linalg/IR/LinalgOps.cpp

```diff
@@ -2494,8 +2496,20 @@ SoftmaxOp::reifyResultShapes(OpBuilder &b,
 void SoftmaxOp::getEffects(
     SmallVectorImpl<SideEffects::EffectInstance<MemoryEffects::Effect>>
         &effects) {
-  getGenericEffectsImpl(effects, getOperation()->getResults(), getDpsInputs(),
-                        getDpsInits());
+  for (auto operand : getDpsInputs()) {
```

Comment:
Why do we need to special case this instead of using `getGenericEffectsImpl`?

Reply:
Because only linalg.softmax does not have the LinalgOp Interface...